cohort-engine/src/main/java/com/ibm/cohort/engine/measure/CDMMeasureEvaluation.java

```diff
@@ -135,9 +153,52 @@ public MeasureReport evaluatePatientMeasure(Measure measure, Context context, St
 		default:
 			// no customizations needed
 		}
+		
+		if(context instanceof DefineContext) {
+			DefineContext defineContext = (DefineContext) context;
+			
+			// Grab the define results from the expression cache
+			if(evidenceOptions.isIncludeDefineEvaluation()) {
+				addDefineEvaluationToReport(report, defineContext);
+			}
+			
+			defineContext.clearExpressionCache();
```

Comment:
I didn't change the signature of the cqf-ruler based code to take a DefineContext (now CDMContext) that would allow this.  I'm curious what you think ContextHelper would look like? Just a class that hides the instanceof check?

Reply:
Basically, yes. It would avoid duplication.